### PR TITLE
Fix If-Modified-Since behavior

### PR DESCRIPTION
**Pull request by mirdhyn**
- `s3api`: passthrough 304 response code from filer, even if `Content-Lenght` is not present
  this bug was leading to 404 responses when passing expired If-Modified-Since   
  
- `filer`: return 304 when `If-Modified-Since` **equal** `Last-Modified`, not only "later"
  as browser saves the time that server returns (using Last-Modified) and uses it with If-Modified-Since on next call
  (note that `volume` handler already behaves this way: https://github.com/chrislusf/seaweedfs/blob/master/weed/server/volume_server_handlers_read.go#L115 )

### CODE REVIEW

**Comment by kmlebedev** [comment]
@mirdhyn  Can you write tests for this?

**Comment by mirdhyn** [comment]
sure, I'll add them shortly.

File: weed/server/filer_server_handlers_read.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func (fs *FilerServer) GetOrHeadHandler(w http.ResponseWriter, r *http.Request,
 		w.Header().Set("Last-Modified", entry.Attr.Mtime.UTC().Format(http.TimeFormat))
 		if r.Header.Get("If-Modified-Since") != "" {
 			if t, parseError := time.Parse(http.TimeFormat, r.Header.Get("If-Modified-Since")); parseError == nil {
-				if t.After(entry.Attr.Mtime) {
+				if t.After(entry.Attr.Mtime) || t.Equal(entry.Attr.Mtime) {
```
Comment (chrislusf) [inline weed/server/filer_server_handlers_read.go:82]: "not before" is simpler

**Comment by chrislusf** [comment]
I will merge this first.

**Comment by mirdhyn** [comment]
Thank you @chrislusf 
@kmlebedev sorry writting tests for that is harder than I expected, especially that there are no similar existing tests that I could rely on :) I will push tests once I figured out how to wire out everything properly.

**Comment by kmlebedev** [comment]
> Thank you @chrislusf
> @kmlebedev sorry writting tests for that is harder than I expected, especially that there are no similar existing tests that I could base on :) I will push tests once I figured out how to wire out everything properly.

@mirdhyn  This is normal, since it is not possible to do such tests right away without rewriting functions into interfaces.

**Pull request merged**